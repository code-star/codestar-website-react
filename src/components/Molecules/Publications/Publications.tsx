import React, { FC } from 'react';
import compose from 'recompose/compose';
import { lifecycle } from 'recompose';
import PublicationCard from '../PublicationCard/PublicationCard';
import { getCachedPublications, IPublication } from "../../../publicationsService";

type PropsInner = {
  publications: IPublication[];
}

type PropsOuter = {}

type Props = PropsInner & PropsOuter;

export const Publications: FC<Props> = ({publications = []}) => {
  const publicationCards = publications.map((p) => <PublicationCard key={p.id} publication={p}></PublicationCard>);
  return (<>{publicationCards}</>);
};

const withUserData = lifecycle({
  async componentDidMount() {
    try {
      const publications = await getCachedPublications();
      this.setState({ publications });
    } catch (err) {
      console.warn('Can\'t set publications, failing silently.');
    }
  }
});

export default compose<Props, PropsOuter>(
  withUserData
)(Publications);
