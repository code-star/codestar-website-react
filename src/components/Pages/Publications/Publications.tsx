import React, { FC } from 'react';
import { translate, TranslationFunction } from 'react-i18next';
import {Fade, Slide, Typography} from '@material-ui/core';
import compose from 'recompose/compose';
import Container from '../../../Container/Container';
import Section from '../../Molecules/Section/Section';
import { withStyles } from '@material-ui/core/styles';
import Publications from '../../Molecules/Publications/Publications';
import { styles } from '../About/About.styles';

interface IPropsInner {
  classes: any;
  t: TranslationFunction;
}

interface IPropsOuter {}

export const PublicationsPage: FC<IPropsInner> = ({ t, classes }) => {
  const [publicationsText0, publicationsText1] = t('PUBLICATIONS_TEXTS', {
    returnObjects: true,
  });
  return (
    <Section scrollname={'publications'}>
      <Container marginTopNavBar>
        <div className="row justify-content-center">
          <div className={`col-12 col-md-8 ${classes.text}`}>
            <Fade in timeout={2000}>
              <div className="mb-4">
                <Typography variant="h2" color="inherit" gutterBottom>
                  {t('PUBLICATIONS_TITLE')}
                </Typography>
                <Typography variant="body1" color="inherit" gutterBottom>
                  {publicationsText0}
                  <a href="https://youtube.com/codestar" className={classes.link}>Codestar YouTube</a>
                  {publicationsText1}
                </Typography>
              </div>
            </Fade>
            <Slide in timeout={1000} direction="up">
              <div>
                <Publications />
              </div>
            </Slide>
          </div>
        </div>
      </Container>
    </Section>
  );
};

export default compose<IPropsInner, IPropsOuter>(
  withStyles(styles),
  translate(['about'], { wait: true })
)(PublicationsPage);
