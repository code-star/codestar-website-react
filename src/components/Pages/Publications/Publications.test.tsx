import React from "react";
import {PublicationsPage} from './Publications';
import renderer from "react-test-renderer";

// Mock for "t", the translate function from react-i18next
const t = (text:string | string[]) => text;

jest.mock('@material-ui/core/Fade', () => "mock-fade");
jest.mock('@material-ui/core/Slide', () => "mock-slide");
jest.mock('../../Molecules/Publications/Publications', () => "mock-publications");

describe('<PublicationsPage />', () => {
  it('renders the publication page', () => {
    const comp = renderer
      .create(
        <PublicationsPage key={1} classes={{}} t={t} />
      )
      .toJSON();
    expect(comp).toMatchSnapshot();
  });
});
