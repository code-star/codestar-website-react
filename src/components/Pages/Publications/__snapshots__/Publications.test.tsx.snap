// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PublicationsPage /> renders the publication page 1`] = `
<section
  className="
			null
			WithTheme(WithWidth(Section))-section"
  name="publications"
>
  <div
    className="WithTheme(WithWidth(Section))-element"
    name="publications"
  />
  <div
    className="
			container
			
			
			
			WithTheme(WithWidth(Container))-marginTopNavBar
			
			
		"
  >
    <div
      className="row justify-content-center"
    >
      <div
        className="col-12 col-md-8 undefined"
      >
        <mock-fade
          in={true}
          timeout={2000}
        >
          <div
            className="mb-4"
          >
            <h2
              className="MuiTypography-root MuiTypography-h2 MuiTypography-colorInherit MuiTypography-gutterBottom"
            >
              PUBLICATIONS_TITLE
            </h2>
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit MuiTypography-gutterBottom"
            >
              P
              <a
                href="https://youtube.com/codestar"
              >
                Codestar YouTube
              </a>
              U
            </p>
          </div>
        </mock-fade>
        <mock-slide
          direction="up"
          in={true}
          timeout={1000}
        >
          <div>
            <mock-publications />
          </div>
        </mock-slide>
      </div>
    </div>
  </div>
</section>
`;
